beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -152,3 +153,10 @@ func (r *RegularSync) validatePendingSlots() error {
 	oldBlockRoots = nil
 	return nil
 }
+
+func (r *RegularSync) clearPendingSlots() {
+	r.pendingQueueLock.RLock()
```

Comment:
should this be a read lock if you're writing?

Reply:
good point, will change it